Rgb/add onboarding documentation (#455)
mozilla/experimenter-docs: docs/onboarding/onboarding-feature-ios.md

The onboarding feature enables experimentating with the 'new user onboarding flow'. The onboarding flow is presented to the user on each new install, and is made up of a series of full screen 'views', referred to as '**cards**'. The purpose of the onboarding flow is to enable the user to configure a small number of app enhancing settings. Each card provides context for each setting and the ability to take an appropratie action, or skip to the next card.

The onboarding feature enables customize each card's:

- title copy
- body copy
- imagery
- button copy
- button actions
- number of buttons (one or two)
- sequencing

# About this document
This document is a guide for staff who wish to configure the new user onboarding flow through the experimenter interface.

It is also a living document:

- the onboarding feature may be under active development
- card types may be added
- card attributes may be added

# Scene setting

The onboarding feature is built on top of Nimbus, Mozilla's experimentation platform. Nimbus allows you to send bits of configuration to application features from Experimenter, the web-application staff use to launch and manage experiments and rollouts.

> Using Experimenter in the general case is documented elsewhere, so this document is specifically concerned with configuring the onboarding feature. [Example experiment](https://experimenter.services.mozilla.com/nimbus/lifestyles-images-onboarding-experiment/summary)

Reviewer: ```suggestion
Using Experimenter in the general case is documented elsewhere, so this document is specifically concerned with configuring the onboarding feature. Example experiment: [Lifestyles images onboarding experiment](https://experimenter.services.mozilla.com/nimbus/lifestyles-images-onboarding-experiment/summary)
```

# References
For the most up-to-date configurations, the main code base will always be the best place to check.
- [Nimbus manifest](https://github.com/mozilla-mobile/firefox-ios/blob/main/nimbus.fml.yaml)

> - [onboarding feature manifest](https://github.com/mozilla-mobile/firefox-ios/blob/main/nimbus-features/onboardingFrameworkFeature.yaml)

Reviewer: ```suggestion
- [Onboarding feature manifest](https://github.com/mozilla-mobile/firefox-ios/blob/main/nimbus-features/onboardingFrameworkFeature.yaml)
```

> ⚠️ The onboarding feature is a **first run experiment**.
> - [for information on first run experiment](https://experimenter.info/mobile-first-run-experiments)

Reviewer: ```suggestion
⚠️ The onboarding feature is a [first run experiment](https://experimenter.info/mobile-first-run-experiments)
```

              "title": "Onboarding/Onboarding.Welcome.Action.v114",
              "action": "next-card",
            },
          },
          "type": "fresh-install",
          "prerequisites": ["ALWAYS"]
      }
   }
}

```
#### Experimenter card config
```json
{
   "cards":{
      "welcome": {
         "title": "A new title",
      }
   }
}

```
#### Output (the merged result of the existing card and experiment card)
```json
   "cards":{
      "welcome": {
         "title": "A new title",
         "body": "Onboarding/Onboarding.Welcome.Description.v114",
         "order": 10,
         "image": "welcome-globe",
         "link": {
            "title": "Onboarding/Onboarding.Welcome.Link.Action.v114",
            "url": "https://www.mozilla.org/privacy/firefox/",
         },
          "buttons": {
            "primary": {
              "title": "Onboarding/Onboarding.Welcome.Action.v114",
              "action": "next-card",
            },
          },
          "type": "fresh-install",
          "prerequisites": ["ALWAYS"]
      }
   }
```

# Feature Definition
The onboarding feature is split into several values:
- conditions
- cards
- dismissable

## Conditions

> The conditions table is a set of triggers, like messaging, that can be used on cards to include or exclude them from appearing. The coditions list is a json of valid JEXL expressions. These expressions can be used in the `prerequisites` or the `disqualifiers` fields in cards. Eg:

Reviewer: ```suggestion
The conditions table is a set of triggers, like messaging, that can be used on cards to include or exclude them from appearing. The conditions list is a json of valid JEXL expressions. These expressions can be used in the `prerequisites` or the `disqualifiers` fields in cards. E.g.:
```

| title  | Free text or StringID | The title of the button          | "Skip"    |
| action | OnboardingActions     | The action the button will take. | next-card |

##### OnboardingActions
| Action                     | Description                                                                         |
|:---------------------------|:------------------------------------------------------------------------------------|
| next-card                  | Will take the user to the next card                                                 |
| sync-sign-in               | Will take the user to the sync sign in flow                                         |
| request-notifications      | Will request to allow notifications from the user                                   |
| set-default-browser        | Will send the user to settings to set Firefox as their default browser              |
| open-default-browser-popup | Will open up a popup with instructions for setting Firefox as their default browser |
| read-privacy-policy        | Will open a webview where the user can read the privacy policy                      |

## Dismissable

> This is a property for the whole onboarding, and controls whether there is an `x` at the top of the screen or not. This is a simple boolean value. ex:

Reviewer: ```suggestion
This is a property for the whole onboarding, and controls whether there is an `x` at the top of the screen or not. This is a simple boolean value:
```

| welcome-ctd      | The welcome image for CTD campaign. |
| sync-devices-ctd | The sync image for CTD campaign. |
| notification-ctd | The notifications image for CTD campaign. |

# Appendix

## Default cards overview

### Default Welcome Card
| Attribute     | Value                                                 |
|---------------|-------------------------------------------------------|
| type          | fresh-install                                         |
| title         | Onboarding/Onboarding.Welcome.Title.v114              |
| body          | Onboarding/Onboarding.Welcome.Description.v114        |
| link          | title: Onboarding/Onboarding.Welcome.Link.Action.v114 |
|               | url: "https://www.mozilla.org/privacy/firefox/"       |
| image         | welcome-globe                                         |
| ordering      | 10                                                    |
| buttons       |                                                       |
| - primary     | title: Onboarding/Onboarding.Welcome.Action.v114      |

> |               | action: next-card                                     |

Reviewer: Should `url:` and `action` be in the first column here? 
<img width="651" alt="image" src="https://github.com/mozilla/experimenter-docs/assets/43795363/61fdd3ad-4de3-4540-8108-1d838659036d">

Reviewer: Same question for `Default Sync card` and `Default Notification card`

Author: Dang it! I mixed up two things. I'll fix it!
